### PR TITLE
PHOENIX-7001: Addendum to initial CDC feature

### PR DESCRIPTION
**Pull request by haridsv**
Addendum to PR #1866

* Revert a change that broke GlobalImmutableTxIndexIT#testLastDDLTimestampOnAsyncIndexes
* Remove ALTER CDC support as it is no longer needed.

### CODE REVIEW

**Review by virajjasani — APPROVED** [review]
+1

**Comment by virajjasani** [comment]
Pending build results https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1899/2/

**Comment by virajjasani** [comment]
Looks like tests are still failing
e.g.
```
java.lang.AssertionError: Ran out of time waiting for index state to become BUILDING last seen actual state is ACTIVE
	at org.junit.Assert.fail(Assert.java:89)
	at org.apache.phoenix.util.TestUtil.waitForIndexState(TestUtil.java:1104)
	at org.apache.phoenix.end2end.index.BaseIndexWithRegionMovesIT.testLastDDLTimestampOnAsyncIndexes(BaseIndexWithRegionMovesIT.java:1712)
```

FYI @tkhurana @palashc could you please also review? Something missing in the tests after the current PR state?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexWithRegionMovesIT.java [hunk 1]
```diff
@@ -1709,6 +1709,7 @@ public void testLastDDLTimestampOnAsyncIndexes() throws Exception {
 
             // run the index MR job.
             IndexToolIT.runIndexTool(false, TestUtil.DEFAULT_SCHEMA_NAME, tableName, indexName);
+            TestUtil.waitForIndexState(conn, fullIndexName, PIndexState.BUILDING);
```
Comment (virajjasani) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexWithRegionMovesIT.java:1712]: It seems this is the problem, it should be `PIndexState.ACTIVE`

Comment (virajjasani) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexWithRegionMovesIT.java:1712]: @haridsv i missed this diff while reviewing, could you please fix this?

Comment (haridsv) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexWithRegionMovesIT.java]: Thanks for catching and sorry about the mistake.

Comment (virajjasani) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexWithRegionMovesIT.java]: No worries at all, Thank you for fixing it

**Comment by virajjasani** [comment]
Now it's failing with NPE, though I am not sure if it is flaky test or it's consistently failing:

https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1899/3/testReport/junit/org.apache.phoenix.end2end.index/GlobalMutableTxIndexIT/testLastDDLTimestampOnAsyncIndexes_GlobalMutableTxIndexIT_transactionProvider_OMID_columnEncoded_false_/

```

java.lang.NullPointerException
	at org.apache.phoenix.end2end.IndexToolIT.verifyMapper(IndexToolIT.java:979)
	at org.apache.phoenix.end2end.IndexToolIT.runIndexTool(IndexToolIT.java:1021)
	at org.apache.phoenix.end2end.IndexToolIT.runIndexTool(IndexToolIT.java:1001)
	at org.apache.phoenix.end2end.IndexToolIT.runIndexTool(IndexToolIT.java:992)
	at org.apache.phoenix.end2end.IndexToolIT.runIndexTool(IndexToolIT.java:986)
	at org.apache.phoenix.end2end.IndexToolIT.runIndexTool(IndexToolIT.java:958)
	at org.apache.phoenix.end2end.index.BaseIndexIT.testLastDDLTimestampOnAsyncIndexes(BaseIndexIT.java:1516)
```

**Comment by haridsv** [comment]
It has to be a flapper as it is working fine on my box (tried 3 times).

image

**Review by virajjasani — APPROVED** [review]

**Pull request merged**